src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreConfiguredInputDataSource.cs

```diff
@@ -58,6 +64,8 @@ protected override IDisposable LinkExternalInput(ITargetBlock<RestoreUpdate> tar
             // and someone is blocked on us on the same thread, the call proceeds
             JoinUpstreamDataSources(source);
 
+            _packageReferenceTelemetryService.LogPackageRestoreEvent(new PackageRestoreTelemetryEvent(PackageRestoreOperationNames.PackageRestoreConfiguredInputDataSourceLinkedToExternalInput, ContainingProject.FullPath));
```

Comment:
Personally I would remove the new event classes. They don't seem to add much value, and require a heap allocation.

Instead, I'd move their constructor arguments to the `Log*Event` method directly, making the call sites simpler too:

```suggestion
            _packageReferenceTelemetryService.LogPackageRestoreEvent(PackageRestoreOperationNames.PackageRestoreConfiguredInputDataSourceLinkedToExternalInput, ContainingProject.FullPath);
```

One reason for using a class is if you expect to add members over time, as that can then be done in a non-API-breaking fashion (i.e. get/set property). However these are internal types so that's not a consideration in this case.

I don't feel too strongly about this. Just something to consider.

Alternatively convert them to `record`s. But I still don't think we need the runtime overhead or the extra API space.

Reply:
I think this is a good idea. This was actually how I started out but thought about the API-breaking possibilities. You're right that this shouldn't be an issue since the API is internal.

---
In reply to: [566551743](https://github.com/dotnet/project-system/pull/6880#discussion_r566551743) [](ancestors = 566551743)